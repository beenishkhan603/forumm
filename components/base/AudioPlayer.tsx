import { ILocalAudioTrack, IRemoteAudioTrack } from 'agora-rtc-sdk-ng'
import React, { useEffect } from 'react'

export interface AudioPlayerProps {
  audioTrack: ILocalAudioTrack | IRemoteAudioTrack | undefined
}

const AudioPlayer = ({ audioTrack }: AudioPlayerProps) => {
  useEffect(() => {
    if (audioTrack) {
      audioTrack?.play()
    }
    return () => {
      audioTrack?.stop()
    }
  }, [audioTrack])

  return <></>
}

export default AudioPlayer
