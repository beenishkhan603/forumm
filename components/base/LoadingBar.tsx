const LoadingBar = () => {
  const shimmerStyle = {
    backgroundSize: '200% 100%',
    animation: 'shimmer 1.5s infinite',
  }

  return (
    <div className="w-full h-2 bg-blue-100 overflow-hidden relative">
      <div
        className="w-full h-full absolute transform translate-x-0 bg-gradient-to-r from-transparent via-blue-500 to-transparent"
        style={shimmerStyle}
      />
    </div>
  )
}

export default LoadingBar
