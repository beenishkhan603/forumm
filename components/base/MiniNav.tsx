import { CreateEventStage } from '@layouts/CreateEventLayout'
import { CreateEventStageInfo } from '@libs/CreateEventContext'
import { useTheme } from '@libs/useTheme'
import { Fragment } from 'react'
import { FaCheck } from 'react-icons/fa'

type GenericSteppedFlow = {
  stage: string
}

export type MiniNavInput = CreateEventStageInfo | GenericSteppedFlow

const MiniNav = ({
  className,
  navItems,
  stage,
  disabled,
  setStage,
}: {
  className?: string
  navItems: {
    label: string
    targetStage: CreateEventStage | string
  }[]
  stage?: MiniNavInput
  disabled?: boolean
  setStage: (data: Partial<MiniNavInput>) => void
}) => {
  const { theme } = useTheme()
  const isLight = theme.type === 'LIGHT'
  const selectedIndex = navItems.findIndex(
    (items) => items.targetStage === stage?.stage
  )
  return (
    <div className={`flex flex-col items-center my-6 py-2 ${className ?? 'px-[10%] sm:pl-[25%] sm:pr-[25%] mt-20'}`}>
    {/* Progress Bar */}
    <div className="relative w-full h-1 bg-gray-300 rounded-full" style={{ maxWidth: '80%' }}>
      {/* Progress fill */}
      <div
        className="absolute top-0 left-0 h-1 bg-black rounded-full transition-all duration-300"
        style={{
          width: `${(selectedIndex / (navItems.length - 1)) * 100}%`, // Calculate the fill width based on the current step
        }}
      ></div>
    </div>
  
    {/* Clickable Labels */}
    <div className="flex w-full justify-between mt-2" style={{ maxWidth: '80%' }}>
      {navItems.map((item, index) => {
        const isHighlighted = index === selectedIndex;
        const isCompleted = index < selectedIndex;
        const { label, targetStage } = item;
  
        return (
          <div
            key={`mininav-label-${item.label}`}
            onClick={() => {
              if (!disabled) setStage({ stage: targetStage });
            }}
            className={`text-sm text-center cursor-pointer transition-all duration-300 ${
              isHighlighted
                ? 'text-black'
                : isCompleted
                ? 'text-gray-500'
                : 'text-gray-400'
            } hover:text-black`}
          >
            {label}
          </div>
        );
      })}
    </div>
  </div>
  







  )
}

export default MiniNav
