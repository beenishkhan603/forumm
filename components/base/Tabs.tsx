import { ReactNode } from 'react'

export const Tab = ({
  children,
  title,
}: {
  children: ReactNode
  title: string
}) => {
  return <>{children}</>
}

export const Tabs = ({
  activeTitle,
  children,
  stickyHeader = false,
}: {
  activeTitle: string
  children: (React.ReactElement | false)[]
  stickyHeader?: boolean
}) => {
  const activeTab: ReactNode | null = children
    .filter((c) => c !== false)
    .find((c) => (c as React.ReactElement).props.title === activeTitle)
  return <>{activeTab}</>
}
