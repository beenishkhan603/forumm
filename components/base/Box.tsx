import { Theme, useTheme } from '@libs/useTheme'
import {
  CSSProperties,
  DetailedHTMLProps,
  HTMLAttributes,
  LegacyRef,
} from 'react'
import LoadingSpinner from './LoadingSpinner'

const Box = (
  props: {
    children?: any
    loading?: boolean
    color?: keyof Theme
    show?: boolean
    ignoreTheme?: boolean
    textColour?: keyof Theme
    blur?: boolean
    innerRef?: LegacyRef<HTMLDivElement>
  } & DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>
) => {
  const { theme } = useTheme()
  const {
    children,
    loading = false,
    color,
    textColour,
    show = true,
    ignoreTheme = false,
    style,
    blur = false,
    innerRef,
    ...divProps
  } = props
  if (!show) {
    return null
  }
  return (
    <div
      {...divProps}
      style={
        ignoreTheme || style
          ? style ?? {}
          : {
            backgroundColor: color && theme ? theme[color] : undefined,
            color:
              theme && textColour
                ? theme[textColour]
                : theme.textColour ?? undefined,
          }
      }
      ref={innerRef}
    >
      {loading ? (
        <Box
          data-test-id={'box-spinner'}
          className="w-full h-full justify-center items-center flex"
        >
          <LoadingSpinner size="medium" />
        </Box>
      ) : blur ? (
        <span style={{ filter: 'blur(2px)' }}>{children}</span>
      ) : (
        children
      )}
    </div>
  )
}

export default Box
