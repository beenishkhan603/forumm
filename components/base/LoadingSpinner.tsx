import { useTheme } from '@libs/useTheme'
import Box from './Box'
import ForumLogoText from '@public/images/ForumLogoText.svg'
import ForumLogoRotating from '@public/images/ForumLogoRotating.svg'


const LoadingSpinner = ({
  className,
  size = 'small',
}: {
  size?: 'small' | 'medium' | 'large'
  className?: string
}) => {
  const { theme } = useTheme()

  const sizeClasses = {
    small: 'w-12 h-12',
    medium: 'w-24 h-24',
    large: 'w-32 h-32',
  }

  return (
    <Box
      data-testid="loading"
      className={`${sizeClasses[size]} ${className ?? ''}`}
      style={{ color: theme.textColour }}
    >
      <ForumLogoRotating fill={theme.textColour} className={`${sizeClasses[size]} ${className ?? ''}` } />
      { (size !== 'small') && (<ForumLogoText fill={theme.textColour} className={`${sizeClasses[size]} ${className ?? ''} !h-12` } />)}
    </Box>
  )
}

export default LoadingSpinner
