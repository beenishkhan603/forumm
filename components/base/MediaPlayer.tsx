import { playOnElement } from '@libs/Agora/util'
import {
  ILocalAudioTrack,
  ILocalVideoTrack,
  IRemoteAudioTrack,
  IRemoteVideoTrack,
} from 'agora-rtc-sdk-ng'
import React, { useEffect, useRef } from 'react'
import Box from './Box'

export interface VideoPlayerProps {
  videoTrack: ILocalVideoTrack | IRemoteVideoTrack | undefined
  audioTrack: ILocalAudioTrack | IRemoteAudioTrack | undefined
  key: string
  fitType?: 'contain' | 'cover' | 'fill'
  className?: string
  children?: React.ReactNode
  mirror?: boolean
  style?: Object
  muteAudio?: boolean
}

const MediaPlayer = ({
  key,
  videoTrack,
  audioTrack,
  children,
  className = 'relative',
  fitType = 'contain',
  mirror = false,
  style = {},
  muteAudio = false,
}: VideoPlayerProps) => {
  const container = useRef<HTMLDivElement>(null)

  useEffect(() => {
    if (!container.current) return
    if (!!videoTrack?.play)
      return videoTrack?.play(container.current, {
        fit: fitType,
        mirror: mirror,
      })

    const teardown = playOnElement(
      videoTrack as unknown as MediaStream,
      container.current
    )

    return () => {
      if (videoTrack?.stop) videoTrack?.stop()
      if (teardown) teardown()
    }
  }, [container, fitType, mirror, videoTrack])

  useEffect(() => {
    if (audioTrack) {
      if (!muteAudio) {
        if (!!audioTrack.play) audioTrack.play()
      } else {
        if (!!audioTrack.stop) audioTrack.stop()
      }
    }
    return () => {
      if (!!audioTrack?.stop) audioTrack?.stop()
    }
  }, [audioTrack, muteAudio])

  return (
    <Box
      key={key}
      innerRef={container}
      className={`${className}`}
      style={style}
    >
      {children}
    </Box>
  )
}

export default MediaPlayer
