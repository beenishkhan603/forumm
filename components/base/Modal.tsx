import { useTheme } from '@libs/useTheme'
import { AnimatePresence, motion } from 'framer-motion'
import React, { useEffect, useRef, useState } from 'react'
import ReactDOM from 'react-dom'
import { IoClose } from 'react-icons/io5'
import Box from './Box'
import Text from './Text'

export interface ModalProps {
  /**
   * Children to enclose in the wrapper.
   */
  children: React.ReactNode
  /**
   * Should the header be shown?
   */
  title?: string

  show: boolean

  setShow?: (show: boolean) => void

  closeButton?: boolean

  animating?: (animating: boolean) => void

  version?: boolean

  className?: string

  bgColor?: string

  zIndex?: string
}

const Modal = ({
  children,
  closeButton = true,
  setShow,
  title,
  show,
  animating,
  version = false,
  className = undefined,
  bgColor = undefined,
  zIndex = '99',
}: ModalProps) => {
  const [mounted, setMounted] = useState(false)
  const [maxHeight, setMaxHeight] = useState(window.innerHeight)
  const ref = useRef<Element | null>(null)
  const { theme } = useTheme()
  const isLight = theme.type === 'LIGHT'

  const handleEsc = (event: KeyboardEvent) => {
    if (event.key === 'Escape') {
      if (setShow) setShow(false)
    }
  }

  useEffect(() => {
    ref.current = document.getElementById('modal-root')
    setMounted(true)
    window.addEventListener('keydown', handleEsc)
    return () => {
      window.removeEventListener('keydown', handleEsc)
    }
  }, [])

  useEffect(() => {
    const handleResize = () => {
      setMaxHeight(window.innerHeight)
    }
    window.addEventListener('resize', handleResize)
    return () => {
      window.removeEventListener('resize', handleResize)
    }
  }, [])

  const toggleAnimation = (override: boolean) => {
    if (!animating) return
    animating(override)
  }

  useEffect(() => {
    if (show) {
      document.body.style.overflow = 'hidden'
    } else {
      document.body.style.overflow = ''
    }

    return () => {
      document.body.style.overflow = ''
    }
  }, [show])

  if (!mounted) {
    return null
  }

  return ReactDOM.createPortal(
    // Started causing a build error out of the blue, works fine.
    // @ts-ignore
    <AnimatePresence>
      {show && (
        <motion.div
          /* istanbul ignore next */
          onAnimationComplete={() => toggleAnimation(false)}
          onAnimationStart={() => toggleAnimation(true)}
          initial={{ backgroundColor: '#00000000' }}
          animate={{ backgroundColor: '#00000080' }}
          id="modal"
          transition={{
            duration: 0.3,
          }}
          exit={{ backgroundColor: '#00000000' }}
          className="fixed w-full h-full top-0 left-0 bg-black z-40 flex items-center justify-center"
          style={{ zIndex }}
          onClick={(e) => {
            e.stopPropagation()
            if (setShow) setShow(false)
          }}
        >
          {!version ? (
            <motion.div
              initial={{
                scale: 0,
                opacity: 0,
              }}
              animate={{
                scale: 1,
                opacity: 1,
              }}
              transition={{
                duration: 0.3,
              }}
              exit={{
                scale: 0,
                opacity: 0,
              }}
              className={`relative flex items-center flex-col top-0 w-full mx-2 xs:mt-[155px] md:mt-2 sm:mx-0 !sm:w-5/6 md:min-w-[90%] lg:min-w-[1000px] xl:min-w-[850px] rounded-2xl p-3 !pt-0 sm:p-6 max-w-xl z-50 !max-h-[calc(100vh-155px)] ${
                !isLight
                  ? 'border border-panel-gray'
                  : 'border border-forumm-menu-border'
              } overflow-y-auto ${className ?? ''}`}
              onClick={(e) => e.stopPropagation()}
              style={{
                background: bgColor || theme.backgroundColour,
              }}
            >
              <Box
                className={`sticky top-0 flex w-full items-center py-3 box-border min-h-16`}
                style={{
                  backgroundColor: theme.backgroundColour,
                  zIndex: 1000,
                }}
              >
                <Text
                  className={`text-center text-lg flex-1 font-semibold`}
                  ignoreTheme
                >
                  {title}
                </Text>
                {closeButton && (
                  <button
                    onClick={(e) => {
                      setShow!(false)
                      e.stopPropagation()
                    }}
                    className="absolute right-0"
                  >
                    <IoClose className="w-8 h-8" />
                  </button>
                )}
              </Box>
              <Box className={`w-full`} ignoreTheme>
                {children}
              </Box>
            </motion.div>
          ) : (
            <motion.div
              initial={{
                scale: 0,
                opacity: 0,
              }}
              animate={{
                scale: 1,
                opacity: 1,
              }}
              transition={{
                duration: 0.3,
              }}
              exit={{
                scale: 0,
                opacity: 0,
              }}
              className={`relative flex items-center flex-col top-0 w-full sm:w-3/4 max-w-xl z-50 rounded-xl overflow-y-auto ${
                className ? className : ''
              }`}
              onClick={(e) => e.stopPropagation()}
              style={{
                background: theme.backgroundColour,
                maxHeight: `${maxHeight}px`,
              }}
            >
              <Box className={`w-full`} ignoreTheme>
                {children}
              </Box>
            </motion.div>
          )}
        </motion.div>
      )}
    </AnimatePresence>,
    ref.current!
  )
}

export default Modal
