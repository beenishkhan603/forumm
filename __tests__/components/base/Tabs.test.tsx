import { Tab, Tabs } from '@components/base/Tabs'
import { render } from '@testing-library/react'

it.each(['1', '2', '3'])('renders tabs unchanged', (activeTitle) => {
  const { container } = render(
    <Tabs activeTitle={activeTitle}>
      <Tab title="1">1</Tab>
      <Tab title="2">2</Tab>
      <Tab title="3">3</Tab>
    </Tabs>
  )
  expect(container).toMatchSnapshot()
})
