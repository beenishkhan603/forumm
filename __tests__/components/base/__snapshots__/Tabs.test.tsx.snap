// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tabs unchanged 1`] = `
<div>
  1
</div>
`;

exports[`renders tabs unchanged 2`] = `
<div>
  2
</div>
`;

exports[`renders tabs unchanged 3`] = `
<div>
  3
</div>
`;
